### PR TITLE
Processing data models

### PR DESCRIPTION
**Pull request by SFarida**
### Project requirements
Be sure to reference the [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png) from the sneak peek if you have any questions about table or column names or data relationships.

- Create a new app. Give it a meaningful name.
- Initialize your project with Git.
- Make sure that your project has Postgres database set up.
- Build your project schema.
    - Create and run the necessary migration files.
    - Table and column names should match the ERD diagram.
        - note: photo for users table should be a link to an image
    - Foreign keys should be included.
        - Pay attention that in `posts` table the column holding the foreign key to the `users` table should be named `author_id`
    - All columns that are foreign keys should have a corresponding index.

### CODE REVIEW

**Review by Qoosim — COMMENTED** [review]
#  *Status: Invalid Request* :red_circle: 

Hi @SFarida, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.



## Invalid Code Review Request

You have submitted a project, that follow wrong workflow. You're expected to follow [Gitflow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md), not Github flow. Kindly check the project requirements to confirm this.
Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: db/migrate/20231028090955_create_likes.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class CreateLikes < ActiveRecord::Migration[7.0]
+  def change
+    create_table :likes do |t|
+      t.references :author, null: false, foreign_key: { to_table: 'users' }
```
Comment (codepantha) [inline db/migrate/20231028090955_create_likes.rb:4]: - [ ] In keeping with the guidelines given in the [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd_v1_1.png), the foreign key should be `user_id` and not `author_id` as you have specified in your migration file above ☝️ 

#### Only `Post` should have `author_id` 👍

File: db/migrate/20231028090921_create_comments.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+class CreateComments < ActiveRecord::Migration[7.0]
+  def change
+    create_table :comments do |t|
+      t.references :author, null: false, foreign_key: { to_table: 'users' }
```
Comment (codepantha) [inline db/migrate/20231028090921_create_comments.rb:4]: - [ ] In keeping with the guidelines given in the [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd_v1_1.png), the foreign key should be `user_id` and not `author_id` as you have specified in your migration file above ☝️ 
### Only `Post` should have `author_id` 👍

**Review by codepantha — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED ♻️ 

## Hi @SFarida 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights 🎯 
✔️ Descriptive PR
✔️ Created migration files to set up database tables 📌 
✔️ Good commit history
✔️ No linter errors 🟢 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- n/a

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: db/migrate/20231028090921_create_comments.rb (hunk 2 above)
Comment (SFarida) [reply db/migrate/20231028090921_create_comments.rb:4]: Hello @codepantha thank you for the review, let me correct this

File: db/migrate/20231104131811_change_author_foreign_key_to_user_id_in_likes.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+class ChangeAuthorForeignKeyToUserIdInLikes < ActiveRecord::Migration[7.0]
+  def change
+    rename_column :likes, :author_id, :user_id
```
Comment (rloterh) [inline db/migrate/20231104131811_change_author_foreign_key_to_user_id_in_likes.rb:3]: - Good job implementing improvements!

Comment (rloterh) [inline db/migrate/20231104131811_change_author_foreign_key_to_user_id_in_likes.rb:3]: - Good job implementing improvements!

File: db/migrate/20231028090921_create_comments.rb (hunk 2 above)
Comment (rloterh) [reply db/migrate/20231028090921_create_comments.rb:4]: well resolved! ✔️

File: db/migrate/20231028090955_create_likes.rb (hunk 1 above)
Comment (rloterh) [reply db/migrate/20231028090955_create_likes.rb:4]: well resolved! ✔️

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi Farida,  

Your project is complete! 🏁 There is nothing else to say other than... it's time to merge it 🍾 Congratulations!  🎉💯🌟 

## Highlights 🎯 
🎲 Good job on introducing improvements to active records migration query!
🎲 ReadMe looks great!

Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**